src/game/client/tf/vgui/tf_classmenu.cpp

```diff
@@ -875,7 +875,11 @@ void CTFClassMenu::OnKeyCodePressed( KeyCode code )
 {
 	m_KeyRepeat.KeyDown( code );
 
-	if ( code > KEY_0 && code <= KEY_9 )
+	if ( code == KEY_0 || code == KEY_PAD_0 ) {
+		SelectClass( iRemapIndexToClass[ TF_CLASS_RANDOM ] );
```

Comment:
```suggestion
		SelectClass( TF_CLASS_RANDOM  );
```

`iRemapIndexToClass` maps the menu index to the TF_CLASS enum. No reason to use it if you're passing `TF_CLASS_RANDOM` directly.